src/Chainweb/SPV/PayloadProof.hs

```diff
@@ -139,8 +159,20 @@ instance (MerkleHashAlgorithm a, MerkleHashAlgorithmName a) => FromJSON (Payload
 data SomePayloadProof where
     SomePayloadProof :: (MerkleHashAlgorithm a, MerkleHashAlgorithmName a) => !(PayloadProof a) -> SomePayloadProof
 
+deriving instance Show SomePayloadProof
+
+instance Eq SomePayloadProof where
+    (SomePayloadProof (a :: PayloadProof aalg)) == (SomePayloadProof (b :: PayloadProof balg))
+        | merkleHashAlgorithmName @aalg == merkleHashAlgorithmName @balg =
+            unsafeCoerce b == a
```

Comment:
isn't there a better way to do this given the witnesses above?

Reply:
The alternative is to use `Data.Type.Equality.testEquality` which would require either 

1. add a proper `Singleton` for `PayloadProof` and define an `TestEquality` instance based on that,
2. move the `unsafeCoerce` into a `TestEquality` instance for `PayloadProof`, or
3. use `typeOf` from `Data.Typeable`.

It's not really important, since the instance is only used for testing (in order to ensure that roundtrips yield same results). I think, I try (3.). If that doesn't work I'll just move the instance out of production code to the test orphans module.

Reply:
The `Data.Typeable` approach kinda leads into a rabbit hole (and generally I try to avoid generic reflection). I think, it is to disruptive for an instance that is only used for testing. I am going to just move it to the test orphans.